apps/avatax/src/modules/avatax/order-cancelled/avatax-order-cancelled-adapter.ts

```diff
@@ -11,11 +15,28 @@ export type AvataxOrderCancelledTarget = VoidTransactionArgs;
 export class AvataxOrderCancelledAdapter implements WebhookAdapter<{ avataxId: string }, void> {
   private logger = createLogger("AvataxOrderCancelledAdapter");
 
+  static AvaTaxOrderCancelledAdapterError = BaseError.subclass("AvaTaxOrderCancelledAdapterError");
+  static DocumentNotFoundError =
+    this.AvaTaxOrderCancelledAdapterError.subclass("DocumentNotFoundError");
+
   constructor(
     private avataxClient: AvataxClient,
     private avataxOrderCancelledPayloadTransformer: AvataxOrderCancelledPayloadTransformer,
   ) {}
 
+  /**
+   * Locally extract error code - but this should be more global, with some normalization/parsing logic/middleware
+   */
+  private extractAvaTaxErrorCode = (error: unknown): string | null => {
+    const parsedError = z.object({ code: z.string() }).safeParse(error);
```

Comment:
**Question:** can't we have enum here and add support for additional error codes like https://github.com/saleor/apps/blob/main/apps/avatax/src/modules/avatax/avatax-errors-parser.ts ?

Reply:
yes!